src/TwitterLoginButton.android.js

```diff
@@ -0,0 +1,100 @@
+import React, {
+  PropTypes,
+} from 'react';
+import {
+  requireNativeComponent,
+  StyleSheet,
+  View,
+} from 'react-native';
+
+
+type Event = Object;
+type TooltipBehaviorIOS = 'auto' | 'force_display' | 'disable';
+
+/**
+ * A button that initiates a log in or log out flow upon tapping.
+ */
+ class TwitterLoginButton extends React.Component {
+   static defaultProps: {
+     style: typeof styles.defaultButtonStyle,
+   };
+
+   props: {
+     /**
+      * Represents the read permissions to request when the login button
+      * is pressed.
+      */
+     readPermissions?: Array<string>,
+
+     /**
+      * Represents the publish permissions to request when the login
+      * button is pressed.
+      */
+     publishPermissions?: Array<string>,
+
+     /**
+      * The callback invoked upon error/completion of a login request.
+      */
+     onLoginFinished?: (error: Object, result: Object) => void,
+
+     /**
+      * The callback invoked upon completion of a logout request.
+      */
+     onLogoutFinished?: () => void,
+
+     /**
+      * View style, if any.
+      */
+     style?: any,
+   };
+
+   _eventHandler(event: Event) {
+   console.log('EVENT_IN_REACT '+event)
+     const eventDict = event.nativeEvent;
+     if (eventDict.type === 'loginFinished') {
+       if (this.props.onLoginFinished){
+         this.props.onLoginFinished(eventDict.error, eventDict.result);
+       }
+     } else if (eventDict.type === 'logoutFinished') {
+       if (this.props.onLogoutFinished) {
+         this.props.onLogoutFinished();
+       }
+     }
+   }
+
+   render() {
+     return (
+       <RCTTwitterLoginButton
```

Comment:
I think it does not matter 'cause this namespace is not visible for library users.